conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -0,0 +1,4506 @@
+     io:println(0.0f / 0.0f); // @output NaN
+
+     a = -0.0f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(-0.0f / 0.0f); // @output NaN
+}
+
+Test-Case: output
+Description: Test division scenarios for different nillable float values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+     float? a = 2.5E-17;
+     float? b = 2.5E+17;
+     float? c = b / a;
+     io:println(c); // @output 1.0000000000000001E34
+
+     a = 1.423223E6;
+     b = 2.34413E2;
+     c = a / b;
+     io:println(c); // @output 6071.433751541083
+
+     a = 25.E1742f;
+     b = 1.0f;
+     c = a / b;
+     io:println(c); // @output Infinity
+
+     a = 25.E1742f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output Infinity
+
+     a = 25.E17f;
+     b = 0.9E-10f;
+     c = a / b;
+     io:println(c); // @output 2.7777777777777776E28
+
+     a = -25.E17f;
+     b = 0.9E-10f;
+     c = a / b;
+     io:println(c); // @output -2.7777777777777776E28
+
+     a = 17.E12f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output Infinity
+
+     a = -17.E12f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output -Infinity
+
+     a = 11.E12f;
+     b = -0.0f;
+     c = a / b;
+     io:println(c); // @output -Infinity
+
+     a = 17.E1290f;
+     b = 13.E1521f;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = 0.0f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = -0.0f;
+     b = 0.0f;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = 11.E12f;
+     b = ();
+     c = a / b;
+     io:println(c.toBalString()); // @output ()
+
+     a = ();
+     b = 13.E1521f;
+     c = a / b;
+     io:println(c.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test division by infinity for different float values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+
+     float a = float:Infinity;
+     float b = float:Infinity;
+     float c = a / b;
+     io:println(c); // @output NaN
+     io:println(float:Infinity / float:Infinity); // @output NaN
+
+     a = 14.34f;
+     b = float:Infinity;
+     c = a / b;
+     io:println(c); // @output 0.0
+     io:println(14.34f / float:Infinity); // @output 0.0
+
+     a = 14.34f;
+     b = -float:Infinity;
+     c = a / b;
+     io:println(c); // @output -0.0
+     io:println(14.34f / -float:Infinity); // @output -0.0
+
+     a = float:NaN;
+     b = float:Infinity;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(float:NaN / float:Infinity); // @output NaN
+
+     a = float:NaN;
+     b = -float:Infinity;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(float:NaN / -float:Infinity); // @output NaN
+}
+
+Test-Case: output
+Description: Test division by infinity for different nillable float values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
```

Comment:
Shall we use `DecimalFloatingPointNumber` instead of `floating-point-literal`?
Don't need `numeric-literal` label? 
Missing `float`, `FloatTypeSuffix`, `float:Infinity`, `float:NaN`, `optional-type`, `numeric-nil-lifting`, `unary-minus`, `nil-literal`, `value:toBalString` labels.